src/ruby/ext/grpc/rb_grpc.c

```diff
@@ -467,6 +467,7 @@ void Init_grpc_c() {
   grpc_rb_mGrpcCore = rb_define_module_under(grpc_rb_mGRPC, "Core");
   grpc_rb_sNewServerRpc = rb_struct_define(
       "NewServerRpc", "method", "host", "deadline", "metadata", "call", NULL);
+  rb_global_variable(&grpc_rb_sStatus);
```

Comment:
does `grpc_rb_sNewServerRpc` also need this?

Also wondering about `grpc_rb_sBatchResult` in `rb_call.c`.

Reply:
No because `rb_struct_define`, like `rb_define_class` all basically all the C API functions that end up creating a class or module, do "root" that object, which means the GC will never collect nor move it.

`grpc_rb_sStatus` need it only because that class is defined in Ruby, and grabbed in C via `rb_const_get`.

Reply:
An alternative could be to define that struct in C like the other (and `Struct.new("Status")` should probably be avoided altogether as it might clash with other gems), but I went for the fix the the minimal amount of changes.